plugins/modules/lambda_event.py

```diff
@@ -252,13 +266,32 @@ def lambda_event_stream(module, client):
     else:
         module.fail_json(msg="Source parameter 'source_arn' is required for stream event notification.")
 
-    # check if optional sub-parameters are valid, if present
-    batch_size = source_params.get("batch_size")
-    if batch_size:
-        try:
-            source_params["batch_size"] = int(batch_size)
-        except ValueError:
-            module.fail_json(msg=f"Source parameter 'batch_size' must be an integer, found: {batch_size}")
+    if state == "present":
+        if module.params["event_source"].lower() == "sqs":
+            # Default 10. For standard queues the max is 10,000. For FIFO queues the max is 10.
+            if not source_params.get("batch_size"):
+                source_params["batch_size"] = 10
+            else:
+                if source_params["source_arn"].endswith(".fifo"):
+                    if source_params["batch_size"] > 10:
+                        module.fail_json(msg="For FIFO queues the maximum batch_size is 10.")
+                    if source_params.get("maximum_batching_window_in_seconds"):
+                        module.fail_json(
+                            msg="maximum_batching_window_in_seconds is not supported by Amazon SQS FIFO event sources."
+                        )
+                else:
+                    if source_params["batch_size"] > 10000 or source_params["batch_size"] < 100:
+                        module.fail_json(msg="For standard queue batch_size must be between 100 and 10000.")
+
+        elif module.params["event_source"].lower() == "stream":
+            if not source_params.get("batch_size"):
+                # Default 100 for streams.
+                source_params["batch_size"] = 100
+            elif source_params["batch_size"] > 10000 or source_params["batch_size"] < 100:
+                module.fail_json(msg="batch_size for streams must be between 100 and 10000")
+
+        if source_params["batch_size"] > 10 and not source_params.get("maximum_batching_window_in_seconds"):
+            source_params["maximum_batching_window_in_seconds"] = 1
```

Comment:
The parameter description says 
``For streams and Amazon SQS event sources, when I(batch_size) is set to a value greater than C(10), I(maximum_batching_window_in_seconds) defaults to C(1).``
therefore this code isn't supposed to move at line ``271`` (to be applied for SQS queue only)?

Reply:
Shouldn't it be applied `For streams and Amazon SQS event sources`?

Reply:
if it applies for these 2 types only this means somewhere we need to run a condition before doing this

Reply:
The module supports only two types of `event_sources` `choices: ['stream', 'sqs']` in any case.